caps/capabilities.go

```diff
@@ -103,6 +106,27 @@ func (r *Repository) Add(cap *Capability) error {
 	return nil
 }
 
+// AddType adds a capability type to the repository.
+// Type names must be unique within the repository.  An error is returned if
+// this constraint is violated. Note that unlike capabilities, types are append
+// only. We don't have any reasonable scenarios where a capability type would
+// have to be removed.
+func (r *Repository) AddType(t Type) error {
+	r.Lock()
```

Comment:
https://github.com/ubuntu-core/snappy/pull/109 just for you

On 17 November 2015 at 18:29, Gustavo Niemeyer <notifications@github.com>
wrote:

> In caps/capabilities.go
> <https://github.com/ubuntu-core/snappy/pull/105#discussion_r45098539>:
>
> > @@ -103,6 +106,27 @@ func (r *Repository) Add(cap *Capability) error {
> >  	return nil
> >  }
> >
> > +// AddType adds a capability type to the repository.
> > +// Type names must be unique within the repository.  An error is returned if
> > +// this constraint is violated. Note that unlike capabilities, types are append
> > +// only. We don't have any reasonable scenarios where a capability type would
> > +// have to be removed.
> > +func (r *Repository) AddType(t Type) error {
> > +	r.Lock()
>
> The concurrency handling of the repository is an internal implementation
> detail, so we shouldn't allow external code to Lock it. As such, this
> should really be under r.m or similar, rather than being an embedded type
> that gives the world access to its methods.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ubuntu-core/snappy/pull/105/files#r45098539>.
>